halyard-core/src/main/java/com/netflix/spinnaker/halyard/core/registry/v1/BillOfMaterials.java

```diff
@@ -48,40 +60,58 @@
     Artifact monitoringDaemon;
     Artifact spinnaker;
 
-    @Data
-    static class Artifact {
-      String version;
-      // TODO(lwander) dependencies will go here.
+    String getArtifactVersion(String artifactName) {
+      return getFieldVersion(Services.class, this, artifactName);
     }
+  }
 
-    String getArtifactVersion(String artifactName) {
-      Optional<Field> field = Arrays.stream(Artifacts.class.getDeclaredFields())
-          .filter(f -> {
-            boolean nameMatches = f.getName().equals(artifactName);
-            boolean propertyMatches = false;
-            JsonProperty property = f.getDeclaredAnnotation(JsonProperty.class);
-            if (property != null) {
-              propertyMatches = property.value().equals(artifactName);
-            }
-            return nameMatches || propertyMatches;
-          })
-          .findFirst();
+  @Data
+  static class Artifact {
+    String version;
+  }
+
+  static private <T> String getFieldVersion(Class<T> clazz, T obj, String artifactName) {
+    Optional<Field> field = Arrays.stream(clazz.getDeclaredFields())
+        .filter(f -> {
+          boolean nameMatches = f.getName().equals(artifactName);
+          boolean propertyMatches = false;
+          JsonProperty property = f.getDeclaredAnnotation(JsonProperty.class);
+          if (property != null) {
+            propertyMatches = property.value().equals(artifactName);
+          }
+          return nameMatches || propertyMatches;
+        })
+        .findFirst();
 
-      if (!field.isPresent()) {
-        throw new RuntimeException("No supported spinnaker artifact named " + artifactName);
-      }
+    try {
+      return ((Artifact) field
+          .orElseThrow(() -> new NoKnownArtifact(artifactName))
+          .get(obj)).getVersion();
+    } catch (IllegalAccessException e) {
+      throw new RuntimeException(e);
+    } catch (NullPointerException e) {
+      throw new RuntimeException("Versioned artifact " + artifactName + " is not listed in the BOM");
+    }
+  }
 
-      try {
-        return ((Artifacts.Artifact) field.get().get(this)).getVersion();
-      } catch (IllegalAccessException e) {
-        throw new RuntimeException(e);
-      } catch (NullPointerException e) {
-        throw new RuntimeException("Spinnaker artifact " + artifactName + " is not listed in the BOM");
-      }
+  static class NoKnownArtifact extends RuntimeException {
+    NoKnownArtifact(String msg) {
+      super(msg);
     }
   }
 
   public String getArtifactVersion(String artifactName) {
-    return services.getArtifactVersion(artifactName);
+    try {
+      return services.getArtifactVersion(artifactName);
+    } catch (NoKnownArtifact ignored) {
+    }
+
+    try {
+      return "0.7.5";
```

Comment:
Sorry, I thought I squashed this. It was for testing since our up-to-date BOMs don't have a `dependencies` section yet.